controlpanel/frontend/views/app.py

```diff
@@ -291,13 +300,16 @@ class AddAdmin(UpdateApp):
     def perform_update(self, **kwargs):
         app = self.get_object()
         user = get_object_or_404(User, pk=self.request.POST['user_id'])
-        userapp = UserApp.objects.create(
-            app=app,
-            user=user,
-            is_admin=True,
-        )
-        userapp.save()
-        messages.success(self.request, f"Granted admin access to {user.name}")
+        if user.auth0_id:
+            userapp = UserApp.objects.create(
+                app=app,
+                user=user,
+                is_admin=True,
+            )
+            userapp.save()
+            messages.success(self.request, f"Granted admin access to {user.name}")
+        else:
+            messages.error(self.request, "Failed to grant admin access to {user.name} because they lack an Auth0 ID.")
```

Comment:
```suggestion
            messages.error(self.request, f"Failed to grant admin access to {user.name} because they lack an Auth0 ID.")
```

Reply:
@davidread aha... I wondered about `f` strings and opted to go for the conservative option. Will revise.

Reply:
OK fixed, squashed and pushed.